src/consensus.rs

```diff
@@ -3,39 +3,57 @@ use std::{
     time::{Duration, Instant},
 };
 
-use raft::{prelude::*, storage::MemStorage};
+use storage::content_manager::toc::TableOfContentRef;
+
+use raft::{eraftpb::Message as RaftMessage, prelude::*};
 
 const TICK_PERIOD_MS: u64 = 100;
 
+type Node = RawNode<TableOfContentRef>;
+
+pub enum Message {
+    FromClient(Vec<u8>),
+    #[allow(dead_code)]
+    FromPeer(Box<RaftMessage>),
+}
+
 pub struct Consensus {
-    node: RawNode<MemStorage>,
+    node: Node,
     receiver: Receiver<Message>,
 }
 
 impl Consensus {
-    pub fn new(logger: &slog::Logger) -> (Self, Sender<Message>) {
+    pub fn new(
+        logger: &slog::Logger,
+        toc_ref: TableOfContentRef,
+    ) -> raft::Result<(Self, Sender<Message>)> {
         let config = Config {
             id: 1,
+            // The max size limits the max size of each appended message. Mostly, 1 MB is enough.
+            max_size_per_msg: 1024 * 1024 * 1024,
```

Comment:
I think it should be `1024 * 1024` for 1MB no?

Reply:
This is actually an unnecessary config at this point, I was just playing with it during debugging. I will remove it.